kernel/fde/fde.go

```diff
@@ -126,3 +136,52 @@ func InitialSetup(runSetupHook RunSetupHookFunc, params *InitialSetupParams) (*I
 	}
 	return res, nil
 }
+
+// CheckFeatures returns the features of fde-setup hook.
+func CheckFeatures(runSetupHook RunSetupHookFunc) ([]string, error) {
+	req := &SetupRequest{
+		Op: "features",
+	}
+	output, err := runSetupHook(req)
+	if err != nil {
+		return nil, err
+	}
+	var res struct {
+		Features []string `json:"features"`
+		Error    string   `json:"error"`
+	}
+	if err := json.Unmarshal(output, &res); err != nil {
+		return nil, fmt.Errorf("cannot parse hook output %q: %v", output, err)
+	}
+	if res.Features == nil && res.Error == "" {
+		return nil, fmt.Errorf(`cannot use hook: neither "features" nor "error" returned`)
+	}
+	if res.Error != "" {
+		return nil, fmt.Errorf("cannot use hook: it returned error: %v", res.Error)
+	}
+	return res.Features, nil
+}
+
+// DeviceSetupParams contains the inputs for the fde-setup hook.
+// The encryption key and the device (partition) are passed in.
+type DeviceSetupParams struct {
+	Key    []byte
+	Device string
```

Comment:
It would be nice to document what form the device spec here is expected to take, from your other PR's I assume that this is the device kernel path i.e. /dev/mmcblk0p1 or whatever, but we are a bit inconsistent (for sometimes good reasons) with "Device" fields not being the full path and just being "mmcblk0p1" for example, so documenting what this is supposed to be would be great